### PR TITLE
moved moab core opperations to function for use elsewhere

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

This is the first stage of getting neutronics testing into the CI and relates to part of issue #584 

I am keen to see if the mb object can be passed to a function and returned in this manner

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Shimwell** [comment]
looks like some tests are needed here

**Comment by Shimwell** [comment]
This code coverage is reporting a drop in coverage

However I noticed that the run_neutronics_tests section in this ci run has an error where the coverage data is not being collected
https://app.circleci.com/pipelines/github/ukaea/paramak/2000/workflows/56e52938-74a5-45cc-a91a-de41a66f43a4/jobs/2198

tests/test_parametric_neutronics/test_Reactor_neutronics.py::test_object_properties::test_export_h5m_without_extension PASSED [100%]Coverage.py warning: No data was collected. (no-data-collected)

**Comment by RemDelaporteMathurin** [comment]
@Shimwell the folder test_neutronics needs an init file.

File: paramak/neutronics_utils.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    Returns:
+        (pymoab Core): A pymoab.core.Core() instance
+        (pymoab tag_handle): A pymoab.core.tag_get_handle() instance
+    """
+
+    try:
+        from pymoab import core, types
+    except ImportError as err:
+        raise err('PyMoab not found, export_h5m method not available')
```
Comment (RemDelaporteMathurin) [inline paramak/neutronics_utils.py:17]: Do we need this here ? These lines will never be hit by the CI. Perhaps a simple import (without try except) is enough ?

File: run_tests.sh [hunk 2]
```diff
@@ -1,3 +1,4 @@
+pytest tests/test_neutronics_utils.py -v --cov=paramak --cov-report term --cov-report xml
 pytest tests/test_utils.py -v --cov=paramak --cov-report term --cov-report xml
```
Comment (RemDelaporteMathurin) [inline run_tests.sh:2]:
```suggestion
pytest tests/test_utils.py -v --cov=paramak --cov-append --cov-report term --cov-report xml
```

`--cov-append` flag missing here

File: .circleci/config.yml [hunk 3]
```diff
@@ -15,6 +15,12 @@ jobs:
           command: |
             pip install -r requirements.txt  # this includes optional packages
             python setup.py install  # this dosen't support the optional packages like pip install .[neutronics] does
+
+      - run:
+          name: run neutronics_utils tests
+          command: 
+            pytest tests/test_neutronics_utils.py -v --cov=paramak --cov-report term --cov-report xml --junitxml=test-reports/junit.xml
+
```
Comment (RemDelaporteMathurin) [inline .circleci/config.yml:23]: I can't comment on the `run utils tests` step, but the `--cov-append` flag is missing there.

File: paramak/neutronics_utils.py (hunk 1 above)
Comment (Shimwell) [reply paramak/neutronics_utils.py:17]: Just I would like the paramak to work for people who don't have pymoab installed so I thought it needed a bit of protection. I moved this section of code from the previous export_h5m statement which used this ImportError protection.

**Comment by Shimwell** [comment]
Ah sorry Remi, I should have committed your suggestions instead of redoing them locally. I wasn't logged in and didn't see the commit button so I didn't realize they were full suggestions till now

File: .circleci/config.yml (hunk 3 above)
Comment (RemDelaporteMathurin) [reply .circleci/config.yml:23]: @Shimwell This has been forgotten apparently. I'll create a PR

Comment (Shimwell) [reply .circleci/config.yml:23]: ah sorry I thought that i did this, sorry for my mistake

**Pull request merged**